crates/client/sync/src/l2.rs

```diff
@@ -111,105 +157,104 @@ pub async fn sync<B, C>(
+                    let bonsai_class = Arc::clone(bonsai_class);
+                    let state_update = Arc::new(state_update);
+                    let state_update_1 = Arc::clone(&state_update);
+
+                    // verify_l2 takes a long time, we don't want to starve the event loop
+                    tokio::task::spawn_blocking(move || {
+                        verify_l2(block_n, &state_update, &overrides, &bonsai_contract, &bonsai_class, block_hash)
+                            .expect("verifying block");
+                    })
+                    .await
+                    .expect("verification task panicked");
+
+                    // hack because tokio tasks need to be 'static
+                    Arc::try_unwrap(state_update_1).expect("arc should not be aliased")
+                };
+
+                tokio::join!(
+                    async {
+                        let block_conv = crate::convert::block(block).await;
+                        block_sender.send(block_conv).await.expect("block reciever channel is closed");
+                    },
+                    async {
+                        // Now send state_update, which moves it. This will be received
+                        // by QueryBlockConsensusDataProvider in deoxys/crates/node/src/service.rs
+                        state_update_sender
+                            .send(StateUpdateWrapper::from(state_update))
+                            .await
+                            .expect("state updater is not running");
+                    },
+                    async {
+                        // do the same to class update
+                        class_sender
+                            .send(ClassUpdateWrapper(class_update))
+                            .await
+                            .expect("class updater is not running");
+                    }
+                );
+
+                create_block(command_sink, &mut last_block_hash).await.expect("creating block");
+
+                block_n += 1;
             }
-        }
-    }
-}
+        } => {},
+    );
 
-async fn fetch_block(
-    client: &SequencerGatewayProvider,
-    block_sender: &Sender<mp_block::Block>,
-    block_number: u64,
-) -> Result<(), String> {
-    let block =
-        client.get_block(BlockId::Number(block_number)).await.map_err(|e| format!("failed to get block: {e}"))?;
+    log::debug!("L2 sync finished :)");
+}
 
-    let block_conv = crate::convert::block(block).await;
-    block_sender.send(block_conv).await.map_err(|e| format!("failed to dispatch block: {e}"))?;
+async fn fetch_block(client: &SequencerGatewayProvider, block_number: u64) -> Result<p::Block, L2SyncError> {
+    let block = client.get_block(BlockId::Number(block_number)).await?;
 
-    Ok(())
+    Ok(block)
 }
 
-pub async fn fetch_genesis_block(config: FetchConfig) -> Result<mp_block::Block, String> {
+// FIXME: I don't know why this is required by the CLI,, this is not used within the sync
```

Comment:
This is and artefact of an older version of the code when this was used to retrieve the head of the chain during initialization, but is since no longer used. You can remove it.